test/cljam/algo/vcf_indexer_test.clj

```diff
@@ -41,77 +61,120 @@
+            (is (= (.min-shift csi) 18))
+            (is (= (.depth csi) 5))
+            (is (= (.loffset csi)
+                   {0 {1 3904},
+                    1 {1 3973},
+                    2 {114556929 4031},
+                    3 {1 4783}}))
+            (is (= (chunks->maps (.bidx csi))
+                   {0 {4681 [{:beg 3904, :end 3973}]},
+                    1 {4681 [{:beg 3973, :end 4031}]},
+                    2 {5118 [{:beg 4031, :end 4783}]},
+                    3 {4681 [{:beg 4783, :end 106168320}]}})))))
+    (testing "bcf"
+      (is (not-throw?
+           (vcf-indexer/create-index test-bcf-complex-file
+                                     tmp-csi-file {:shift 14 :depth 5})))
+
+      (do (vcf-indexer/create-index test-bcf-complex-file
+                                    tmp-csi-file {:shift 14 :depth 5})
+          (let [computed ^CSI (csi/read-index tmp-csi-file)
+                csi ^CSI (csi/read-index test-bcf-complex-csi-file)]
+            (is (= (.n-ref csi) (.n-ref computed)))
+            (is (= (.depth csi) (.depth computed)))
+            (is (= (.aux csi) (.aux computed)))
+            (is (= (.loffset csi) (.loffset computed)))
+            (is (= (.bidx csi) (.bidx computed))))))))
 
 (deftest about-vcf-changed-chr-order
-  (with-before-after {:before (prepare-cache!)
-                      :after (clean-cache!)}
-    (do (vcf-indexer/create-index test-vcf-changed-chr-order-file
-                                  tmp-csi-file {:shift 18 :depth 5})
-        (let [csi ^CSI (csi/read-index tmp-csi-file)]
-          (is (= ["19" "20" "X"] (:chrs (.aux csi))))
-          (is (= (chunks->maps (.bidx csi))
-                 {0 {4681 [{:beg 2096, :end 2205}]},
-                  1 {4681 [{:beg 1647, :end 1811}],
-                     4685 [{:beg 1811, :end 2096}]},
-                  2 {4681 [{:beg 2205, :end 70975488}]}}))
-          (is (= (.loffset csi)
-                 {0 {1 2096},
-                  1 {1 1647, 1048577 1811},
-                  2 {1 2205}}))))))
+  (testing "vcf"
+    (with-before-after {:before (prepare-cache!)
+                        :after (clean-cache!)}
+      (do (vcf-indexer/create-index test-vcf-changed-chr-order-file
+                                    tmp-csi-file {:shift 18 :depth 5})
+          (let [csi ^CSI (csi/read-index tmp-csi-file)]
+            (is (= ["19" "20" "X"] (:chrs (.aux csi))))
+            (is (= (chunks->maps (.bidx csi))
+                   {0 {4681 [{:beg 2096, :end 2205}]},
+                    1 {4681 [{:beg 1647, :end 1811}],
+                       4685 [{:beg 1811, :end 2096}]},
+                    2 {4681 [{:beg 2205, :end 70975488}]}}))
+            (is (= (.loffset csi)
+                   {0 {1 2096},
+                    1 {1 1647, 1048577 1811},
+                    2 {1 2205}}))))))
+  (testing "bcf"
+    (with-open [r (vcf/reader test-bcf-changed-chr-order-file)]
+      (let [file-offsets (vcf/read-file-offsets r)
+            computed
```

Comment:
Add a type hint `^CSI` to the declaration of `computed`, and you won't need to type hint every subsequent use-site of the name. Try not to use type hints too much and keep your code concise 💪 The same goes for some of the test cases below.